android/src/main/java/com/mobeye/geolocation/MyLocation.java

```diff
@@ -9,14 +11,14 @@
 
 public class MyLocation {
 
-    public MyLocation(){}
-
+    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR2)
     public MyLocation(Location l) {
         this.provider = l.getProvider();
         this.latitude = l.getLatitude();
         this.longitude = l.getLongitude();
         this.accuracy = l.getAccuracy();
         this.time = l.getTime();
+        this.mock = l.isFromMockProvider();
```

Comment:
Ah mais dans la doc ils disent :
_This method was deprecated in API level 31.
Prefer isMock() instead._

Du coup il suffit pas d'appeler isMock à la place ? 
j'ai pas bien compris ce que ça veut dire dans la description de isMock : 
_If this location comes from the Android framework, this indicates that the location was provided by a test location provider, and thus may not be related to the actual location of the device._
mais sinon ça semble correspondre